src/main/java/de/hpi/bpt/argos/eventProcessing/mapping/EventEntityMapperImpl.java

```diff
@@ -0,0 +1,122 @@
+import de.hpi.bpt.argos.storage.dataModel.entity.Entity;
+import de.hpi.bpt.argos.storage.dataModel.event.Event;
+import de.hpi.bpt.argos.storage.dataModel.event.type.EventType;
+import de.hpi.bpt.argos.storage.dataModel.mapping.EventEntityMapping;
+import de.hpi.bpt.argos.storage.dataModel.mapping.MappingCondition;
+import de.hpi.bpt.argos.util.LoggerUtilImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EventEntityMapperImpl extends ObservableImpl<EventMappingObserver> implements EventEntityMapper {
+	private static final Logger logger = LoggerFactory.getLogger(EventEntityMapperImpl.class);
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void setup(Observable<EventCreationObserver> eventReceiver) {
+		eventReceiver.subscribe(this);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void onEventCreated(EventEntityMappingStatus mappingStatus,
+			EventType eventType, List<TypeAttribute> eventTypeAttributes, Event event, List<Attribute> eventAttributes) {
+
+		if (mappingStatus.isMapped()) {
+			notifyObservers((EventMappingObserver observer) ->
+					observer.onEventMapped(event, mappingStatus.getEventOwner(), mappingStatus.getUsedMapping()));
+			return;
+		}
+
+		List<EventEntityMapping> mappings = PersistenceAdapterImpl.getInstance().getEventEntityMappingsForEventType(eventType.getId());
+
+		for (EventEntityMapping mapping : mappings) {
+			List<MappingCondition> mappingConditions = PersistenceAdapterImpl.getInstance().getMappingConditions(mapping.getId());
+
+			if (mappingConditions.isEmpty()) {
+				// no mapping condition, so we can not map this
+				continue;
+			}
+
+			List<TypeAttribute> entityTypeAttributes = PersistenceAdapterImpl.getInstance().getTypeAttributes(mapping.getEntityTypeId());
+
+			try {
+				StringBuilder sqlQuery = new StringBuilder();
+				sqlQuery.append("FROM EntityImpl entity WHERE");
+
+				sqlQuery.append(String.format(" entity.%1$s = %2$s",
+						getAttributeName(mappingConditions.get(0).getEntityTypeAttributeId(), entityTypeAttributes),
+						getAttributeValue(mappingConditions.get(0).getEventTypeAttributeId(), eventAttributes)));
+
+				for (int i = 1; i < mappingConditions.size(); i++) {
+					sqlQuery.append(String.format(" AND entity.%1$s = %2$s",
+							getAttributeName(mappingConditions.get(i).getEntityTypeAttributeId(), entityTypeAttributes),
+							getAttributeValue(mappingConditions.get(i).getEventTypeAttributeId(), eventAttributes)));
+				}
+
+				Entity owner = PersistenceAdapterImpl.getInstance().getMappingEntity(sqlQuery.toString());
+				event.setEntityId(owner.getId());
+				mappingStatus.setEventOwner(owner, mapping);
+
+				notifyObservers((EventMappingObserver observer) -> observer.onEventMapped(event, owner, mapping));
+
+				return;
+
+			} catch (EventEntityMappingException e) {
+				LoggerUtilImpl.getInstance().error(logger, e.getMessage(), e);
```

Comment:
This will not log every mapping. This is just in case some attributes could not be found even though they should exist.